## This is the Jupyter Notebook for the final project of my Data Science Immesive at General Assembly Austin Tx

#### Temporary Hypothesis, which features of my dataset best predict time on trail. 
## What sort of things can I add to this dataset? Weather,etc..
My data was obtained through the Appalachian Trail Conservancy. I called and asked for all availible data on Thru hikers. Talk about form. 

In [131]:
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [132]:
at = pd.read_excel('BenHikersInfo.xlsx')

In [133]:
at.shape

(17835, 28)

In [134]:
def clean(df):
    df.drop(df.index[[17264,17236]], inplace = True)
    df['Sex'] = df['Sex'].map({'M':0,'F':1})
    df['Finish'] = pd.to_datetime(df['Finish'])
    df['Start'] = pd.to_datetime(df['Start'])
    df.drop(labels = df[['Disease Other','School','DogWeightLbs','MinorityComments','School',
                         'DogBreed','Disability','Under2KAmount']], axis = 1, inplace = True)
    df.rename(columns = {'AgeYrFin': 'Age', 'Days Elapsed':'Days_On_Trail','Race or Heritage': 'Race',
                        'StAbbrev':'State', 'Hiking Disability ': 'Hiking_Disability', 
                         'Hike Type': 'Hike_Type', 'Yr Comp':'completion_yr', 'Hiking Disability':'Hiking_Disability'},inplace = True)
    df.dropna(subset= ['Start','Sex','Race','Country','Days_On_Trail','State','Hike_Type'],inplace = True)
    df['completion_yr'].fillna(2016,inplace = True)
    df['completion_yr'] = df['completion_yr'].astype(int)
    df['Days_On_Trail'] = df['Days_On_Trail'].astype(int)
    df['Age'] = df['Age'].fillna(df['Age'].mean())
    
    
clean(at)

In [135]:
at.head()

Finish        Age                Race State  Sex Country Hike_Type  \
0  2016-09-28  27.000000  Caucasian or White    NH  1.0     USA   Section   
8  1948-12-31  36.252632   Prefer not to say    PA  0.0     USA      NOBO   
9  1951-10-10  28.000000   Prefer not to say    CT  0.0     USA      SOBO   
10 1951-09-30  36.252632   Prefer not to say    GA  0.0     USA      NOBO   
20 1959-09-17  20.000000  Caucasian or White    GA  0.0     USA      NOBO   

        Start  Days_On_Trail  completion_yr    ...    HikedWDog WinterHike  \
0  2015-03-18            561           2016    ...        False      False   
8  1948-01-01            366           1948    ...        False      False   
9  1951-05-30            134           1951    ...        False      False   
10 1951-05-31            123           1951    ...        False      False   
20 1959-05-23            118           1959    ...        False      False   

   Under2KSpent SchoolCredit Minority Vegetarian DehydrateOwnFood Giardia  \
0         False        False    False      False            False   False   
8         False        False    False      False            False   False   
9         False        False    False      False            False   False   
10        False        False    False      False            False   False   
20        False        False    False      False            False   False   

     Lyme Norovirus  
0   False     False  
8   False     False  
9   False     False  
10  False     False  
20  False     False  

[5 rows x 21 columns]

In [138]:
at['Hike_Type'].value_counts()


NOBO           10522
Section         3099
SOBO            1503
Flip             914
Alternative      102
alternative        4
Alterntive         1
Yo Yo              1
Name: Hike_Type, dtype: int64

In [75]:
x.value_counts()

0    12381
1     3765
Name: Sex, dtype: int64

In [39]:
at['Minority'].value_counts()

False    16068
True        78
Name: Minority, dtype: int64

In [35]:
at['Sex'].value_counts()

M    12381
F     3765
Name: Sex, dtype: int64

In [ ]:
at[at['Sex'].str.contains('othe')]

In [ ]:
at[at['Sex'].str.contains('Oher')]

In [ ]:
df.drop([17264,17236])

In [ ]:
x['Sex'].str.contains('othe')

In [ ]:
at.shape

In [ ]:
at.isnull().sum().nlargest(30)

In [ ]:
#plt.figure(figsize = (10,10))
#null.plot(kind = 'bar')

# Notes on How I have cleaned my DataSet

1.) 

2.) Convert the Start and Finish Dates to Pandas Datetime Objects

3.) Drop columns with null values that exceed 90 percent of the column 

4.) Drop Dog Weight Column 16238 of possible 17835 values were 0. So nearly the same as null 

5.) Drop Under2kAmount column 16125 of 17835 values were 0. It was just not self reported very well

4.) Rename Columns with spaces in them or w/ confusing names. 

5.) Replace Null Values in yr completed with 2016 since we do know his start date* 

6.) Drop oher and othe out of the sex column

7.) Convert null values in age column to mean. I checked out a distplot for how this would change
with using mean,median or mode, and not alot of change. 

8.)

9.)

10.)

11.)

12.)

13.)